src/project.rs

```diff
@@ -0,0 +1,62 @@
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+use rocksdb::DB;
+use walkdir::{WalkDir, DirEntry};
+
+// TODO test this in /tests as an integration test
+pub fn guess_type(mut p: PathBuf) -> ProjectType {
+    // TODO: make this better...
+    p.push("Cargo.toml");
+    if p.exists() {
+        ProjectType::Rust
+    } else {
+        ProjectType::Go
+    }
+}
+
+pub enum ProjectType {
+    Rust,
+    Go,
+}
+
+pub struct Project {
+    files: Vec<File>
+}
+
+pub struct File {
+    path: String,
+    lines: i32,
+}
+
+
+pub fn analyze_go(path: &mut PathBuf, db: Arc<DB>) {
```

Comment:
Maybe we should start to make a generic interface for a project language? It appears like we will want to allow certain operations for a specific language. We can do a match statement on the project type and then the compiler can do the work of plugging in the capability to perform a certain action (based on whether we have defined an `impl` for that project type).